opentracing-metrics-prometheus-spring-autoconfigure/src/main/java/io/opentracing/contrib/metrics/prometheus/spring/autoconfigure/PrometheusMetricsReporterConfiguration.java

```diff
@@ -13,37 +13,42 @@
  */
 package io.opentracing.contrib.metrics.prometheus.spring.autoconfigure;
 
-import java.util.Set;
-
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.opentracing.contrib.metrics.MetricLabel;
+import io.opentracing.contrib.metrics.MetricsReporter;
+import io.opentracing.contrib.metrics.micrometer.MicrometerMetricsReporter;
+import io.prometheus.client.CollectorRegistry;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import io.opentracing.contrib.metrics.MetricLabel;
-import io.opentracing.contrib.metrics.MetricsReporter;
-import io.opentracing.contrib.metrics.prometheus.PrometheusMetricsReporter;
-import io.prometheus.client.CollectorRegistry;
+import java.util.Set;
 
 @Configuration
 public class PrometheusMetricsReporterConfiguration {
+    @Autowired(required=false)
+    private Set<MetricLabel> metricLabels;
 
     @Autowired
     private CollectorRegistry collectorRegistry;
 
     @Value("${OPENTRACING_METRICS_NAME:}")
     private String metricsName;
 
-    @Autowired(required=false)
-    private Set<MetricLabel> metricLabels;
-
     @Bean
-    public MetricsReporter prometheusMetricsReporter() {
-        PrometheusMetricsReporter.Builder builder = PrometheusMetricsReporter.newMetricsReporter();
+    public MetricsReporter prometheusMetricsReporter(PrometheusMeterRegistry prometheusMeterRegistry) {
```

Comment:
Think this is ok for now - but we might need to look at the two usecases:
1) where no micrometer/prometheus is used in an app - so the auto config is initializing everything
2) where micrometer/prom is already being used by another part of app or lib, and this auto config is just being used to initialize the metrics reporter with the tracer observer.